dll/win32/iphlpapi/iphlpapi_main.c

```diff
@@ -2620,12 +2621,113 @@ SetIpForwardEntryToStack(PMIB_IPFORWARDROW pRoute)
 }
 
 DWORD WINAPI
-NhGetInterfaceNameFromDeviceGuid(DWORD dwUnknown1,
-                                 DWORD dwUnknown2,
-                                 DWORD dwUnknown3,
+NhGetInterfaceNameFromDeviceGuid(GUID * pInterfaceGUID,
+                                 PWCHAR pInterfaceName,
+                                 PULONG pOutBufLen,
                                  DWORD dwUnknown4,
                                  DWORD dwUnknown5)
 {
-    FIXME("NhGetInterfaceNameFromDeviceGuid() stub\n");
-    return 0L;
+    UNICODE_STRING GuidString;
+    WCHAR szKeyName[2*MAX_PATH];
+    HKEY hKey;
+    DWORD result, type;
+
+    if (pInterfaceGUID == NULL || pOutBufLen == NULL)
+    {
+        return ERROR_INVALID_PARAMETER;
+    }
+
+    if (RtlStringFromGUID(pInterfaceGUID, &GuidString) != STATUS_SUCCESS)
```

Comment:
`if (!NT_SUCCESS(RtlStringFromGuid(blabla))`

Reply:
This applies to most of the ifs here: I tend to find "Var = Function(...); if (Var == x)" cleaner than "if (Function(...) == x)" in 99% of cases. YMMV.